src/particle_filter/include/particle_filter/distanceTransform.h

```diff
@@ -0,0 +1,177 @@
+#ifndef DISTANCE_TRANSFORM_H
+#define DISTANCE_TRANSFORM_H
+#include <limits>
+
+class distanceTransform
+{
+ public:
+  double voxel_size; //voxel size
+  double world_range[3][2]; //work space range
+  double ***dist_transform;
+  double ***obstacle_map;
+  distanceTransform(double range, double voxel_size);
+  void build(double cube_para[3], double World_Range[3][2]);
+ protected:
+  int voxel_nums[3]; // voxel number along each dimension
+  void distanceTransform_1D(double ***dist_transform_1D, int range[3], double ***cost_fun, int dir, int idx1, int idx2);
```

Comment:
***dist_transform_1D is a pointer to a pointer to a pointer....yuck.
This can be simplified. possibly with std::vector